lib/python/ship_callsign.py

```diff
@@ -1,42 +1,70 @@
 #!/usr/bin/env python
 # -*- coding: utf-8 -*-
 """
-Reads the ship_callsign text file in the same directory and returns a
-dictionnary of callsign and vessel names.
-Used for SOOP
+Find the ship_callsign from poolparty platform vocab xml file stored in
+content.aodn.org.au and returns the approriate vessel name if exists. If the
+callsign cannot be found, returns an empty string
 
 How to use:
     import os, sys
     sys.path.insert(0, os.path.join(os.environ.get('DATA_SERVICES_DIR'), 'lib'))
-    from python.ship_callsign import ship_callsign_list
+    from python.ship_callsign import *
+
+
+    ship = ship_callsign_list()
+    print ship_callsign('VRDU8')
 
 author : Besnard, Laurent
 """
 
-import os
-from ConfigParser import SafeConfigParser
+import urllib2
+import xml.etree.ElementTree as ET
 
+def platform_code_list():
+    platform_vocab_url = 'http://content.aodn.org.au/IMOS/Vocabularies/aodn_aodn-platform-vocabulary_version-1-0.rdf'
+    response           = urllib2.urlopen(platform_vocab_url)
+    html               = response.read()
+    root               = ET.fromstring(html)
+    next_item_exist    = True
+    platform           = {}
+    item               = 0
 
-def _call_parser(conf_file):
-    parser             = SafeConfigParser()
-    parser.optionxform = str  # to preserve case
-    parser.read(conf_file)
-    return parser
+    while next_item_exist:
```

Comment:
This probably works but there's a much more straight-forward way to iterate over a list in Python:
```python
for item in root:
    if 'Description' in item.tag:
        # etc...
```
or 
```python
for item in root.iter():
```
depending on whether you just want the direct children of root or everything below it (https://docs.python.org/2/library/xml.etree.elementtree.html#xml.etree.ElementTree.Element.iter)

Then you don't need `next_item_exist` or the try/except at the end of the loop.